internal/output/result_test.go

```diff
@@ -40,13 +51,24 @@ func Test_mapIDsToGroupedSARIFFinding(t *testing.T) {
 	t.Parallel()
 
 	tests := []struct {
-		name     string
-		args     models.VulnerabilityResults
-		wantPath string
+		name string
+		args models.VulnerabilityResults
+		want testutility.Snapshot
 	}{
 		{
-			args:     testutility.LoadJSONFixture[models.VulnerabilityResults](t, "fixtures/test-vuln-results-a.json"),
-			wantPath: "fixtures/test-vuln-results-a-grouped.json",
+			args: testutility.LoadJSONFixtureWithWindowsReplacements[models.VulnerabilityResults](t,
+				"fixtures/test-vuln-results-a.json",
+				map[string]string{
+					"/path/to/sub-rust-project/Cargo.lock": "D:\\\\path\\\\to\\\\sub-rust-project\\\\Cargo.lock",
+					"/path/to/go.mod":                      "D:\\\\path\\\\to\\\\go.mod",
+				},
+			),
+			want: testutility.NewSnapshot().WithWindowsReplacements(
```

Comment:
There seems to be a lot of these path replacements that's very similar. 

Let's change this to take a map of  regex's replacements instead of strings,  this way we can add a "default" regex replace rule that covers the majority of the path replacements.

Reply:
I'll explore this but my concern is these are typically pretty hard to craft since most folks only have one OS on hand so I think it's better to go the "boring but long" route than "complex but short" especially since these are tests.

But I'll explore it first to see how complex it actually ends up being before I officially take a stance 🙂

Reply:
Yeah so playing with this locally I'm not sure if this is worth the extra complexity, as you still need to specify the full path - you've effectively trading vertical length for horizontal length.

fwiw the design of this btw is that while yes you'll need to specify the exact line, that is what you can just copy & paste from the snapshot line without much thinking (again, perfect imo because people don't have multiple OSs so they'll be checking the diff from CI); if you were using a regexp you'd have to think more about how you could normalize things and the overlap, and if you mucked up a regexp (or two) it would be harder to see which line is the problem.

I think we should stick with the current layout for now, and revisit after living with it for a bit to see if it is as troublesome as the complexity

Reply:
Makes sense to me, let's stick with this then.